### PR TITLE
Various Shell Script Fixes and Improvements

### PR DESCRIPTION
**Pull request by Androbin**

### CODE REVIEW

**Comment by bazel-io** [comment]
Can one of the admins verify this patch?

**Comment by Androbin** [comment]
@damienmg Continuing #3162

File: third_party/ijar/test/ijar_test.sh [hunk 1]
```diff
@@ -41,7 +41,7 @@ shift
 source ${DIR}/testenv.sh || { echo "testenv.sh not found!" >&2; exit 1; }
 
 function cleanup() {
-  rm -fr "$TEST_TMPDIR"/*
+  rm -fr "${TEST_TMPDIR:?}"/*
```
Comment (damienmg) [inline third_party/ijar/test/ijar_test.sh]: replace ? with ___do__not__exists___ or something like that.

**Comment by damienmg** [comment]
jenkins: test this please

File: third_party/ijar/test/ijar_test.sh (hunk 1 above)
Comment (Androbin) [reply third_party/ijar/test/ijar_test.sh]: How about `"${TEST_TMPDIR:-sentinel}"` ?

**Comment by damienmg** [comment]
SGTM

**Comment by Androbin** [comment]
I will split into 2 PR as requested and link to them here.

**Comment by damienmg** [comment]
Thanks!

**Comment by Androbin** [comment]
Part One: #4051

**Comment by Androbin** [comment]
Part Two: #4052

**Comment by damienmg** [comment]
Closing this one, let'scontinue the review on the other PRs

**Pull request closed without merging**